code/player/TTTPlayer.Flashlight.cs

```diff
@@ -0,0 +1,172 @@
+    {
+        private Flashlight _worldFlashlight;
+        private Flashlight _viewFlashlight;
+        private KeyframeEntity _flashlightHolder;
+
+        public bool HasFlashlightEntity
+        {
+            get
+            {
+                if (IsLocalPawn)
+                {
+                    return _viewFlashlight != null && _viewFlashlight.IsValid();
+                }
+
+                return _worldFlashlight != null && _worldFlashlight.IsValid();
+            }
+        }
+
+        public bool IsFlashlightOn
+        {
+            get
+            {
+                if (IsLocalPawn)
+                {
+                    return HasFlashlightEntity && _viewFlashlight.Enabled;
+                }
+
+                return HasFlashlightEntity && _worldFlashlight.Enabled;
+            }
+        }
+
+        public void ToggleFlashlight()
+        {
+            ShowFlashlight(!IsFlashlightOn);
+        }
+
+        public void ShowFlashlight(bool shouldShow, bool playSounds = true)
+        {
+            if (IsFlashlightOn == shouldShow)
+            {
+                return;
+            }
+
+            if (IsFlashlightOn)
+            {
+                if (IsServer)
+                {
+                    _worldFlashlight.TurnOff();
+                }
+                else
+                {
+                    _viewFlashlight.TurnOff();
+                }
+            }
+
+            if (IsServer)
+            {
+                ClientShowFlashlightLocal(To.Single(this), shouldShow);
+            }
+
+            if (shouldShow)
+            {
+                if (!HasFlashlightEntity)
+                {
+                    if (IsServer)
+                    {
+                        _worldFlashlight = new Flashlight();
+                        _worldFlashlight.EnableHideInFirstPerson = true;
+                        _worldFlashlight.Rotation = EyeRot;
+                        _worldFlashlight.Position = EyePos + EyeRot.Forward * 40f;
+                        _worldFlashlight.SetParent(this);
+                    }
+                    else
+                    {
+                        _flashlightHolder = new KeyframeEntity();
+                        _flashlightHolder.Position = EyePos + EyeRot.Forward * 40f;
+                        _flashlightHolder.Rotation = EyeRot;
+                        //_flashlightHolder.SetParent(this);
+
+                        _viewFlashlight = new Flashlight();
+                        _viewFlashlight.EnableViewmodelRendering = false;
+                        _viewFlashlight.Position = EyePos + EyeRot.Forward * 40f;
+                        _viewFlashlight.Rotation = EyeRot;
+                        _viewFlashlight.SetParent(_flashlightHolder);
+                    }
+                }
+                else
+                {
+                    if (IsServer)
+                    {
+                        // TODO: This is a weird hack to make sure the rotation is right.
```

Comment:
This doesn't seem like much of a TODO, what's the goal in the future here?

Reply:
To remove this if it was fixed ^^